src/hooks/useFetch.ts

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useState } from "react";
+import { OrderItemProps } from "interfaces/Order";
+
+interface sendRequestProps {
+  url: string;
+  method?: string;
+  headers?: {};
+  body?: OrderItemProps | null;
+}
+
+const useFetch = () => {
```

Comment:
props for the custom hook

Reply:
As an extra, I recommend taking a look at RTK Query and rewriting the fetch logic using it